Formula/j/jsontoolkit.rb

```diff
@@ -0,0 +1,98 @@
+class Jsontoolkit < Formula
+  desc "Swiss-army knife library for expressive JSON programming in modern C++"
+  homepage "https://jsontoolkit.sourcemeta.com/"
+  url "https://github.com/sourcemeta/jsontoolkit/archive/refs/tags/v2.0.0.tar.gz"
+  sha256 "00f82f02166beabec80522e2bbc7b839ee9b7ccb631411c42e6fab65186e80ba"
+  license "AGPL-3.0-only"
+
+  depends_on "cmake" => :build
+
+  def install
+    args = %W[
+      -DCMAKE_COMPILE_WARNING_AS_ERROR=OFF
+      -DJSONTOOLKIT_CONTRIB=OFF
+      -DJSONTOOLKIT_TESTS=OFF
+      -DJSONTOOLKIT_DOCS=OFF
+      -DBUILD_SHARED_LIBS=ON
+      -DCMAKE_INSTALL_RPATH=#{rpath}
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"CMakeLists.txt").write <<~EOS
+      cmake_minimum_required(VERSION 3.15)
+      project(JsonToolkitTest)
+      set(CMAKE_CXX_STANDARD 20)
+
+      find_package(JSONToolkit REQUIRED)
+
+      add_executable(test test.cpp)
+      target_include_directories(test PRIVATE #{include})
+      target_link_libraries(test PUBLIC sourcemeta::jsontoolkit::json)
+      target_link_libraries(test PUBLIC sourcemeta::jsontoolkit::jsonschema)
+      target_link_libraries(test PUBLIC sourcemeta::jsontoolkit::jsonpointer)
+      target_link_libraries(test PUBLIC sourcemeta::jsontoolkit::jsonl)
+    EOS
+
+    (testpath/"test.cpp").write <<~EOS
+      #include <sourcemeta/jsontoolkit/json.h>
+      #include <sourcemeta/jsontoolkit/jsonl.h>
+      #include <sourcemeta/jsontoolkit/jsonpointer.h>
+      #include <sourcemeta/jsontoolkit/jsonschema.h>
+
+      #include <cassert>
+      #include <iostream>
+      #include <sstream>
+
+      int main() {
+        // JSON
+        std::istringstream input{R"([ { "foo": 1 }, { "bar": 2 } ])"};
+        const sourcemeta::jsontoolkit::JSON json_doc =
+            sourcemeta::jsontoolkit::parse(input);
+        assert(json_doc.is_array());
+
+        // JSON Pointer
+        const sourcemeta::jsontoolkit::Pointer pointer{1, "bar"};
+        const sourcemeta::jsontoolkit::JSON &value{
+            sourcemeta::jsontoolkit::get(json_doc, pointer)};
+        assert(value.is_integer());
+        assert(value.to_integer() == 2);
+
+        // JSONL
+        std::istringstream jsonl_input{
+            "{ \"foo\": 1 }\n{ \"bar\": 2 }\n{ \"baz\": 3 }"};
+        for (const auto &document : sourcemeta::jsontoolkit::JSONL{jsonl_input}) {
+          assert(document.is_object());
+        }
+
+        // JSON Schema
+        const sourcemeta::jsontoolkit::JSON schema =
+            sourcemeta::jsontoolkit::parse(R"JSON({
+          "$schema": "https://json-schema.org/draft/2020-12/schema",
+          "type": "string"
+        })JSON");
+
+        const auto schema_template{sourcemeta::jsontoolkit::compile(
```

Comment:
The schema compiler only officially supports Draft 4 for now, not 2020-12. Can you set a Draft 4 metaschema instead?

Reply:
cc @tony-go